### PR TITLE
1931 - after successful login the saved language of the user is set

### LINKED ISSUE #1931: 🚀 [Group] [Feature]  Language setting for the user
## :rocket: Group - Feature
#  Language setting for the user 
after successful login the saved language of the user must set

# DB: User.language 

- at registration
 - [x] When registering the user the set language must be saved  #1929

- and in the user settings
- [x] the user can change his language setting to the user setting  #1930

[x] after successful login the saved language of the user is set

### PR DESCRIPTION
**Pull request by ogerly**
## 🍰 Pullrequest
after successful login the saved language of the user is set

### Issues GROUP
- fixes #1931

### CODE REVIEW

File: webapp/store/auth.js [hunk 1]
```diff
@@ -46,6 +46,9 @@ export const getters = {
   termsAndConditionsAgreed(state) {
     return state.user && state.user.termsAndConditionsAgreedVersion === VERSION
   },
+  locale(state) {
+    return state.locale
```
Comment (mattwr18) [inline webapp/store/auth.js]: should we have a fallback here @ogerly ? most of the getters I have seen in or store have `|| null`, etc...

File: webapp/pages/login.vue [hunk 2]
```diff
@@ -7,19 +7,30 @@
...
   asyncData({ store, redirect }) {
     if (store.getters['auth/user'].termsAndConditionsAgreedVersion === VERSION) {
       redirect('/')
     }
   },
   methods: {
     handleSuccess() {
+      const currentLocale = this.$i18n.locale()
+      const userSettingLocale = this.user.locale
+      if (currentLocale !== userSettingLocale) {
```
Comment (roschaefer) [inline webapp/pages/login.vue]: I think the `if` statement is quite unnecessary here, isn't it?

File: webapp/pages/login.vue [hunk 3]
```diff
@@ -7,19 +7,30 @@
...
+        this.$i18n.set(userSettingLocale)
+      }
```
Comment (roschaefer) [inline webapp/pages/login.vue]: How about just setting:
```js
        this.$i18n.set(this.user.locale)
```
And that's it?

File: webapp/store/auth.js (hunk 1 above)
Comment (roschaefer) [reply webapp/store/auth.js]: @ogerly I think you can remove this. You're consuming `this.user.locale` so this code here is never used.

File: webapp/pages/login.vue [hunk 4]
```diff
@@ -7,19 +7,26 @@
...
+  },
   asyncData({ store, redirect }) {
     if (store.getters['auth/user'].termsAndConditionsAgreedVersion === VERSION) {
       redirect('/')
     }
   },
   methods: {
-    handleSuccess() {
+    handleSuccess() {      
+      this.$i18n.set(this.user.locale)
```
Comment (roschaefer) [inline webapp/pages/login.vue]: What about a default here?
```suggestion
      this.$i18n.set(this.user.locale || 'en')   
```

Comment (mattwr18) [reply webapp/pages/login.vue]: @ogerly can you please add this?

**Comment by ogerly** [comment]
sorry

**Pull request merged**